inject/src/main/java/io/micronaut/inject/ParametrizedProvider.java

```diff
@@ -1,47 +0,0 @@
-/*
- * Copyright 2017-2020 original authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- * https://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package io.micronaut.inject;
-
-import jakarta.inject.Provider;
-
-import java.util.Map;
-
-/**
- * An extended version of the {@link Provider} interface for {@link ParametrizedBeanFactory}.
- *
- * @param <T> The type
- * @author graemerocher
- * @since 1.0
- */
-public interface ParametrizedProvider<T> extends Provider<T> {
```

Comment:
this is a breaking change we can't remove it even if it is not used until next major

Reply:
@graemerocher And how to be? Will this class be present in all releases up to version 5.0 now?

As far as I understand, you forgot to remove it when removed the `ParametrizedBeanFactory` class.

Reply:
Anyway I rolled back this change